include/cbmqtm/cbmqtm_dbdistributor.php

```diff
@@ -32,30 +32,34 @@ private static function setDB() {
 		static::$db = new PearDatabase();
 	}
 
-	public function sendMessage($channel, $producer, $consumer, $type, $share, $sequence, $expires, $deliverafter, $userid, $information) {
+	public function sendMessage($channel, $producer, $consumer, $type, $share, $sequence, $expires, $deliverafter, $userid, $information, $deliverRange=array('deliverStartTime'=>'', 'deliverEndTime'=>'', 'canSendOnSaturday'=>'', 'canSendOnSunday'=>'')) {
 		if ($share != '1:M' && $share != 'P:S') {
 			$share = '1:M';
 		}
 		if ($share == '1:M' || !$this->subscriptionExist($channel, $producer, $consumer)) {
-			$this->insertMsg($channel, $producer, $consumer, $type, $share, $sequence, $expires, $deliverafter, $userid, $information);
+			$this->insertMsg($channel, $producer, $consumer, $type, $share, $sequence, $expires, $deliverafter, $userid, $information, $deliverRange);
 		} else {
 			self::setDB();
 			$subrs = static::$db->pquery('select * from cb_mqsubscriptions where channel=?', array($channel));
 			while ($subscriber = static::$db->fetch_array($subrs)) {
-				$this->insertMsg($channel, $producer, $subscriber['consumer'], $type, $share, $sequence, $expires, $deliverafter, $userid, $information);
+				$this->insertMsg($channel, $producer, $subscriber['consumer'], $type, $share, $sequence, $expires, $deliverafter, $userid, $information, $deliverRange);
 			}
 		}
 	}
 
-	private function insertMsg($channel, $producer, $consumer, $type, $share, $sequence, $expires, $deliverafter, $userid, $information) {
+	private function insertMsg($channel, $producer, $consumer, $type, $share, $sequence, $expires, $deliverafter, $userid, $information, $deliverRange) {
 		$rightnow = time();
 		if (empty($deliverafter)) {
 			$deliverafter = 0;
 		}
+		if (empty($deliverRange)) {
```

Comment:
if you get a partial array this "empty" will not apply but you won't get the other array elements initialized (as explained above).

Also, the initialization and the query below are wrong because you are forcing the deliver range start and end which is not the case. Most messages on the queue will not have this restriction and can be delivered at any hour of the day, any day. The start-end, and day conditions must apply only to those messages that require it.

I think you have to do the same "if" code you did for Saturday/Sunday for the start-end